### PR TITLE
Issue 1018 psa scipy bugfix

### LINKED ISSUE #1018: `PSAnalysis.cluster()` in `MDAnalysis.analysis.psa` breaks due to change in kwargs in scipy `dendrogram()` function
## Expected behaviour

Calling `PSAnalysis.plot(...)` after calling `PSAnalysis.run(...)` should generate a heatmap-dendrogram plot showing the chosen clustering according to the selected path metric.  

## Actual behaviour

`PSAnalysis.plot(...)` causes a `TypeError` with the following output:

```bash
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-af029f7dc01a> in <module>()
----> 1 psa_short.plot(filename='dh_ward_psa-short.pdf', linkage='ward');

/nfs/homes3/sseyler/.virtualenvs/research_stable/lib/python2.7/site-packages/MDAnalysis/analysis/psa.py in plot(self, filename, linkage, count_sort, distance_sort, figsize, labelsize)
   1468                                 method=linkage,                                 \
   1469                                 count_sort=count_sort,                          \
-> 1470                                 distance_sort=distance_sort)
   1471         rowidx = colidx = dgram['leaves'] # get row-wise ordering from clustering
   1472         ax_dgram.invert_yaxis() # Place origin at up left (from low left)

/nfs/homes3/sseyler/.virtualenvs/research_stable/lib/python2.7/site-packages/MDAnalysis/analysis/psa.py in cluster(self, distArray, method, count_sort, distance_sort, no_plot, no_labels, color_threshold)
   1738                            count_sort=count_sort, distance_sort=distance_sort,  \
   1739                            no_plot=no_plot, color_threshold=color_threshold,    \
-> 1740                            color_list=color_list)
   1741         return Z, dgram
   1742 

TypeError: dendrogram() got an unexpected keyword argument 'color_list'
```

## Code to reproduce the behaviour

The behavior can be most easily reproduced by running the jupyter notebooks provided in the [PSAnalysisTutorial](https://github.com/Becksteinlab/PSAnalysisTutorial) (e.g., _psa\_short.ipynb_)

## Current version of MDAnalysis: 0.15.0

(This bug should be a problem with any version of MDAnalysis that includes the PSA module.)

## Assessment of the problem

As of SciPy 0.17.0, `scipy.cluster.hierarchy.dendrogram` no longer takes the `color_list` keyword. Although  `color_list` is not explicitly described in the parameters docs, it can be seen in the list of kwargs in the `dendrogram` function signature. For instance:

### SciPy 0.16.1

scipy.cluster.hierarchy.dendrogram(Z, p=30, truncate_mode=None, color_threshold=None, get_leaves=True, orientation='top', labels=None, count_sort=False, distance_sort=False, show_leaf_counts=True, no_plot=False, no_labels=False, color_list=None, leaf_font_size=None, leaf_rotation=None, leaf_label_func=None, no_leaves=False, show_contracted=False, link_color_func=None, ax=None, above_threshold_color='b')

### SciPy 0.17.0

scipy.cluster.hierarchy.dendrogram(Z, p=30, truncate_mode=None, color_threshold=None, get_leaves=True, orientation='top', labels=None, count_sort=False, distance_sort=False, show_leaf_counts=True, no_plot=False, no_labels=False, leaf_font_size=None, leaf_rotation=None, leaf_label_func=None, show_contracted=False, link_color_func=None, ax=None, above_threshold_color='b')

### PR DESCRIPTION
**Pull request by sseyler**
Fixes #1018 

Changes made in this Pull Request:
 - Removed `color_list` kwarg in `scipy.cluster.hierarchy.dendrogram` in `PSAnalysis.cluster`
 - Removed import of `get_map` from `brewer2mpl` package
 - Removed call to `get_map`
 - Changed `orientation` kwarg from "right" to "left" in `scipy.cluster.hierarchy.dendrogram` in `PSAnalysis.cluster` (orientation was reversed).

PR Checklist
------------
 - [x] Tests: 
   Fixes the TypeError produced when calling `PSAnalysis.plot`. Tested by running `psa_short.ipynb` using latest MDAnalysis and SciPy versions.

 - [x] Docs—no updates needed.
 - [x] CHANGELOG updated.
 - [x] Issue raised/referenced.

### CODE REVIEW

**Comment by orbeckst** [comment]
Can you add a simple test that just checks that the dendrogram can be produced? If this is about producing a figure, then just check that a matplotlib figure or axes object can be produced.

File: testsuite/MDAnalysisTests/analysis/test_psa.py [hunk 1]
```diff
@@ -72,3 +75,7 @@ def test_reversal_hausdorff(self):
     def test_reversal_frechet(self):
         err_msg = "Frechet distances did not increase after path reversal"
         assert_(self.frech_matrix[1,2] >= self.frech_matrix[0,1], err_msg)
+
+    def check_dendrogram_produced(self):
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/test_psa.py]: Method name needs to start with `test_` or it is not run by nose.

**Comment by orbeckst** [comment]
@sseyler the minimal tests fail https://travis-ci.org/MDAnalysis/mdanalysis/jobs/165943444 – you need to mark the test as skip when matplotlib is not installed.

**Comment by orbeckst** [comment]
Sean, can you please check why Travis failed and address/discuss?

I'm currently on a phone and it's awful to review Travis reports on mobile.

**Comment by orbeckst** [comment]
Apparently you need to import `module_not_found` ; it's probably defined at the tests top level but just look around, it's one of our functions.

**Comment by orbeckst** [comment]
@sseyler can you please make the changes so that we can finish the PR?

**Comment by orbeckst** [comment]
One travis task failed but that looked like one of the "too many open files" errors so I restarted the task. Let's see... probably all good to merge but I'd like to have a proper green checkmark.

**Comment by sseyler** [comment]
@orbeckst I can add a few more tests to improve the coverage (if that needs to be done). Any suggestions?

**Comment by orbeckst** [comment]
Thanks. More tests are always good, of course!

**Comment by orbeckst** [comment]
ARGH – I somehow merged into master. I should have checked the patch more carefully. Sorry.

@MDAnalysis/coredevs – I will need to rewrite history on master to get rid of 58a091ebd1c2e3c22f601a32965c5daa1b1b616d

We will then open a new PR against develop and see if the patch still applies.

**Comment by orbeckst** [comment]
I forecefully removed the 58a091e commit from master (required temporarily unprotecting the branch). Sorry for this, next time I will double check where the merge wants to go.

The new PR against develop is #1024. I needed to resolve conflicts with @tylerjereddy 's new tests but they passed locally and now waiting for Travis (and then someone else to merge).

**Pull request merged**